src/v/cluster/partition.cc

```diff
@@ -804,6 +804,45 @@ ss::future<> partition::remove_persistent_state() {
     }
 }
 
+/**
+ * Return the index of this node in the list of voters, or nullopt if it
+ * is not a voter.
+ */
+static std::optional<size_t>
+voter_position(raft::vnode self, const raft::group_configuration& raft_config) {
+    const auto& voters = raft_config.current_config().voters;
+    auto position = std::find(voters.begin(), voters.end(), self);
+    if (position == voters.end()) {
+        return std::nullopt;
+    } else {
+        return position - voters.begin();
+    }
+}
+
+// To reduce redundant re-uploads in the typical case of all replicas
+// being alive, have all non-0th replicas delay before attempting to
+// reconcile the manifest. This is just a best-effort thing, it is
+// still okay for them to step on each other: finalization is best
+// effort and the worst case outcome is to leave behind a few orphan
+// objects if writes were ongoing while deletion happened.
+static ss::future<bool> maybe_finalize(
```

Comment:
Naming: usually a `maybe_` function will "maybe do the thing" rather than just returning the bool for whether the thing should be done.

We can simplify this a bit in anticipation of tombstones: in this forthcoming code (https://github.com/jcsp/redpanda/commit/fbaa6614242eebb5c71cdc03346268df0d1a876c) the erase logic changes to only attempt erase on the first voter.  So rather than this being a flag for whether to finalize, and passing that flag into erase(), we can just make this function a "should_cleanup_remote", and if it returns false, don't both calling any finalize or erase stuff.